core/src/main/java/org/testcontainers/images/RemoteDockerImage.java

```diff
@@ -95,4 +92,21 @@ protected final String resolve() {
             throw new ContainerFetchException("Failed to get Docker client for " + imageName, e);
         }
     }
+
+    private DockerImageName getImageName() throws InterruptedException, ExecutionException {
+        return imageNameFuture.get();
+    }
+
+    @ToString.Include(name = "imageName", rank = 1)
+    private String imageNameToString() {
+        if (!imageNameFuture.isDone()) {
+            return "<resolving>";
+        }
+
+        try {
+            return getImageName().toString();
```

Comment:
Why is try necessary after the future is done?

Reply:
you need to get the value from the future. Unless I miss something 😅